### DISCUSSION TITLE
Error: P1001: Can't reach database server at 'IP NAS SYNOLOGY'

### DISCUSSION THREAD

**Discussion by Salsito48** [prisma/prisma#24593] [Q&A] [answered]
### Question

Hello, I have a problem connecting prisma with my database hosted on a physical NAS through maridb in phpmyadmin, below I leave the prisma code to see if you can help me -->

.env -->
```
DATABASE_URL="mysql://login:********@********:8306/accessdb"
```
schema.prisma -->

```prisma
datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

model User {
  id String @id @default(cuid())
  name String
}
```
This error happens to me when running 'npx prisma db push' command to connect my database

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @Salsito48 👋 

Can you verify that you can connect to your MariaDB database using the credentials and connection string provided in the `DATABASE_URL` using a Database GUI like [TablePlus](https://tableplus.com/)? If you have special characters in your password, you will need to [percent encode](https://www.prisma.io/docs/orm/reference/connection-urls#special-characters) them.

**Reply by Salsito48** [reply]
Yes and it connects with the database perfectly

**Reply by ludralph** [reply]
Are the special characters in the password?

**Reply by Salsito48** [reply]
just numbers and a plus symbol (+). Even so, encoding the special characters gives me the same error. I even tried to copy the table plus address and paste it, I made the corresponding modifications and it didn't work, it gave me the same error P1001

**Reply by Salsito48** [reply, marked as answer]
I have managed to solve the error, I would simply have to add connect_timeout=300 and it would be solved
